lightly/models/modules/heads.py

```diff
@@ -198,19 +198,36 @@ class SimCLRProjectionHead(ProjectionHead):
     "We use a MLP with one hidden layer to obtain zi = g(h) = W_2 * σ(W_1 * h)
     where σ is a ReLU non-linearity." [0]
 
-    [0] SimCLR, 2020, https://arxiv.org/abs/2002.05709
+    "We use a 3-layer MLP projection head on top of a ResNet encoder." [1]
 
+    [0] SimCLR, 2020, https://arxiv.org/abs/2002.05709
+    [1] SimCLR, 2020, https://arxiv.org/abs/2006.10029
     """
 
     def __init__(
-        self, input_dim: int = 2048, hidden_dim: int = 2048, output_dim: int = 128
+        self,
+        input_dim: int = 2048,
+        hidden_dim: int = 2048,
+        output_dim: int = 2048,
+        num_layers: int = 3,
     ):
-        super(SimCLRProjectionHead, self).__init__(
-            [
-                (input_dim, hidden_dim, None, nn.ReLU()),
-                (hidden_dim, output_dim, None, None),
-            ]
-        )
+        """Initialize a new SimCLRProjectionHead instance.
+
+        Args:
+            input_dim: Number of input dimensions.
+            hidden_dim: Number of hidden dimensions.
+            output_dim: Number of output dimensions.
+            num_layers: Number of hidden layers.
+        """
+        layers: list[tuple[int, int, Optional[nn.Module], Optional[nn.Module]]] = []
+        layers.append((input_dim, hidden_dim, nn.BatchNorm1d(hidden_dim), nn.ReLU()))
```

Comment:
This changes the logic, SimCLR v1 doesn't have batchnorm in the projection head. Maybe it would be easier to add a `SimCLRV2ProjectionHead` if there are significant differences between the two versions.

Reply:
Are you sure SimCLR v1 doesn't use batchnorm? I see it right here: https://github.com/google-research/simclr/blob/1.0/model_util.py#L112

The v1 and v2 papers don't mention batchnorm, but the source code still uses it for all layers.

Reply:
You are right, we must have missed this. Interesting that neither vissl nor solo-learn add batchnorm either:
* https://github.com/facebookresearch/vissl/blob/04788de934b39278326331f7a4396e03e85f6e55/configs/config/pretrain/simclr/simclr_4node_resnet.yaml#L53
* https://github.com/vturrisi/solo-learn/blob/6a891ee9af6e28ae071313b596f8e553b7227b64/solo/methods/simclr.py#L48-L52

The issue is that we cannot change it globally right now as it breaks some other code. I guess we have to add a `batch_norm: bool = True` argument for now and set it to `False` in those places:
* https://github.com/lightly-ai/lightly/blob/527572e2bef6d4df4b96dbe3c2a5e46e36ae1528/lightly/models/simclr.py#L35
* https://github.com/lightly-ai/lightly/blob/527572e2bef6d4df4b96dbe3c2a5e46e36ae1528/lightly/cli/_cli_simclr.py#L24

Reply:
> The issue is that we cannot change it globally right now as it breaks some other code.

Is that other code incorrect? Happy to fix this in other locations too if necessary. Can also set the number of layers to 2 by default if you want to maintain backwards compatibility.

Reply:
Found at least one test where I can set `batch_norm=False`, not sure if there are others. We could change the default to False if we need to, but we should change it to True someday once we've re-trained with the extra layers.

Reply:
> Found at least one test where I can set batch_norm=False, not sure if there are others.

We also have to set `batch_norm=False` in these two places:
* https://github.com/lightly-ai/lightly/blob/527572e2bef6d4df4b96dbe3c2a5e46e36ae1528/lightly/models/simclr.py#L35
* https://github.com/lightly-ai/lightly/blob/527572e2bef6d4df4b96dbe3c2a5e46e36ae1528/lightly/cli/_cli_simclr.py#L24

I believe if you change this the tests should pass.

> We could change the default to False if we need to, but we should change it to True someday once we've re-trained with the extra layers.

I think setting it to True by default makes sense. We just have to set it to False in the places where we need backwards compatibility.